### PR TITLE
feat(resources index):  add total number of evaluations of every learning units resource

### PR DESCRIPTION
**Pull request by PabloAbara**
# Base PR
- PR based on main branch. 

# Context
- With this PR, we modify the resource#index action adding the 'number of evaluations' field. 

# Changelog
- Add new model method 'number_of_evaluations' on Resource model
- Call model method on endpoint resources#index
- Modify Rspec test on number_of_evaluations and average_evaluation (that was not implemented). 

# QA
- Launch Backend App
- Enter to localhost:3000
- Go to http://localhost:3000/api/learning_units/1/resources
- Check if it returns the corresponding average_evaluation and number_of_evaluations.

### CODE REVIEW

File: spec/requests/api/resources_spec.rb [hunk 1]
```diff
@@ -18,20 +18,28 @@
...
+                   url: { type: :string },
+                   average_evaluation: { type: :string },
+                   number_of_evaluations: { type: :integer }
                  }
                },
                required: %w[id name]
-        run_test!
+        run_test! do |response|
+          data = JSON.parse(response.body)
+          expect(data[0]['average_evaluation']).to eq('3.0')
```
Comment (AlehandroL) [inline spec/requests/api/resources_spec.rb:40]: Magic number: in the 'before' block you use the default factory value, is best practice to specify in the same file the value that you will be expecting, even if the factory provides one (i.e. create(:resource_evaluation, user:, evaluation: 3) )

**Review by AlehandroL — CHANGES_REQUESTED** [review]
Looks great. Please run rswag before merging.

File: spec/requests/api/resources_spec.rb (hunk 1 above)
Comment (PabloAbara) [reply spec/requests/api/resources_spec.rb:40]: Done, thx.

**Review by AlehandroL — APPROVED** [review]

**Review by jjsantos01 — APPROVED** [review]
Great job!
Tested on local and worked ok

**Review by jjsantos01 — APPROVED** [review]

**Pull request merged**